Meadowlark-js/packages/meadowlark-core/src/validation/PaginationValidator.ts

```diff
@@ -0,0 +1,48 @@
+/* eslint-disable no-use-before-define */ // function declarations are hoisted properly in ES6
```

Comment:
@bradbanister as you know I like private local functions. What do you think about tweaking the eslint file to allow this pattern, assuming it is possible?

My understanding is that function expressions are no longer hoisted in ES6, so they wouldn't be appropriate. But regular function declarations are still hoisted and thus perfectly safe.

Reply:
I actually like that eslint rule because it makes code browsing easier. You always know which direction to scroll when referring to a called function. Could you relocate them to the beginning of the function? Alternatively, I'd stick with the eslint disable, but do it per-line rather than whole file.

Reply:
Thanks to hoisting, that too will work 😄 . In C#, the private functions must be at the end if they are accessing variables from outside their scope. But in JS/TS this is perfectly fine.

Personally I have a small preference for them at the end of the function, so that I can quickly scan the "heart" of the function without needing to look past the private function definitions. But that is unusual, and I have definitely heard some complaints / confusion about that. So let's try out putting them at the top.